compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -277,6 +278,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             }
         };
 
+        // All other patterns constitute a read, which causes us to diverge
+        // if the type is never.
+        if ty.is_never() && !matches!(pat.kind, PatKind::Wild | PatKind::Never | PatKind::Or(_)) {
```

Comment:
same here, not diverging is always correct, incorrectly diverging is unsound. So please exhaustively match the "does diverge" variants